### PR TITLE
Code optimization.

### PR DESCRIPTION
**Pull request by whatdoyouseedoc**
Http methods was moved to separated file.
Refactored isCorrectHttpMethod.
Optimized httpRx methods assignment.

### CODE REVIEW

File: lib/rx-http.js [hunk 1]
```diff
@@ -2,8 +2,9 @@ const httpRx = {};
 const request = require('request');
 const Observable = require('rxjs').Observable;
 const HTTP_STATUS_CODES = require('./http-status-codes');
+const HTTP_METHODS = require('./http-methods');
 
-const isCorrectHttpMethod = method => /get|head|patch|post|put|delete/i.test(method);
+const isCorrectHttpMethod = method => RegExp(HTTP_METHODS.join('|'), 'i').test(method);
```
Comment (JasonRammoray) [inline lib/rx-http.js]: I propose to move `isCorrectHttpMethod` to another file and cover with unit tests (6 positive and 1 negative).

File: lib/check-http-method.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const HTTP_METHODS = require('./http-methods');
+
+/**
+ * Checks is http method correct
```
Comment (JasonRammoray) [inline lib/check-http-method.js]: Checks if http method is correct or not

File: lib/check-http-method.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+ * @param {string} method - Method for checking
```
Comment (JasonRammoray) [inline lib/check-http-method.js]: string -> String

File: lib/check-http-method.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ * @return {boolean} Is correct or not
```
Comment (JasonRammoray) [inline lib/check-http-method.js]: boolean -> Boolean

File: lib/rx-http.js [hunk 5]
```diff
@@ -2,11 +2,12 @@ const httpRx = {};
 const request = require('request');
 const Observable = require('rxjs').Observable;
 const HTTP_STATUS_CODES = require('./http-status-codes');
+const HTTP_METHODS = require('./http-methods');
+const checkHttpMethod = require('./check-http-method');
```
Comment (JasonRammoray) [inline lib/rx-http.js]: I propose to keep an original name of that variable in order to reflect semantic.
"Is<something>" implies a logical outcome (true or false).

File: tests/check-http-method.spec.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const HTTP_METHODS = require('../lib/http-methods');
+const checkHttpMethod = require('../lib/check-http-method');
+
+describe('checkHttpMethod function ->', () => {
+    HTTP_METHODS.forEach(method => {
+        it(`should checked as true for "${method}" method`, done => {
```
Comment (JasonRammoray) [inline tests/check-http-method.spec.js]: should treat HTTP method "${method}" as a correct method

File: tests/check-http-method.spec.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+            const result = checkHttpMethod(method);
+            expect(result).toBe(true);
+            done();
+        });
+    });
+
+    [null, undefined, NaN, 'wrong_method', 42].forEach(method => {
+        it(`should checked as false for incorrect "${method}" method`, done => {
```
Comment (JasonRammoray) [inline tests/check-http-method.spec.js]: I was thinking about string)
It is unlikely, that sane developer is going to add `null`, `undefined` or `NaN` to http methods.

Comment (JasonRammoray) [reply tests/check-http-method.spec.js]: And I propose to change spec name to 'should be able to detect an incorrect http method'.

File: lib/rx-http.js (hunk 5 above)
Comment (whatdoyouseedoc) [reply lib/rx-http.js]: Do you mean it should be like
```
const isCorrectHttpMethod = require('./check-http-method');
```
?

File: lib/check-http-method.js (hunk 2 above)
Comment (whatdoyouseedoc) [reply lib/check-http-method.js]: +

File: lib/check-http-method.js (hunk 3 above)
Comment (whatdoyouseedoc) [reply lib/check-http-method.js]: +

File: lib/check-http-method.js (hunk 4 above)
Comment (whatdoyouseedoc) [reply lib/check-http-method.js]: +

File: tests/check-http-method.spec.js (hunk 6 above)
Comment (whatdoyouseedoc) [reply tests/check-http-method.spec.js]: +

File: tests/check-http-method.spec.js (hunk 7 above)
Comment (whatdoyouseedoc) [reply tests/check-http-method.spec.js]: +

File: lib/rx-http.js (hunk 5 above)
Comment (JasonRammoray) [reply lib/rx-http.js]: Yep.

Comment (whatdoyouseedoc) [reply lib/rx-http.js]: Done.

File: tests/check-http-method.spec.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+const HTTP_METHODS = require('../lib/http-methods');
+const checkHttpMethod = require('../lib/check-http-method');
+
+describe('checkHttpMethod function ->', () => {
+    HTTP_METHODS.forEach(method => {
+        it(`should treat HTTP method "${method}" as a correct method`, done => {
+            const result = checkHttpMethod(method);
+            expect(result).toBe(true);
+            done();
```
Comment (JasonRammoray) [inline tests/check-http-method.spec.js]: I am not sure why we are using `done` callback in this case.
`checkHttpMethod` works synchronously.
I've used `done` callback in the previous unit tests, because request -> response cycle is asynchronous in it's nature.

File: tests/check-http-method.spec.js [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+        });
+    });
+
+    it(`should be able to detect an incorrect http method`, done => {
+        const result = checkHttpMethod('incorrect_method');
+        expect(result).toBe(false);
+        done();
```
Comment (JasonRammoray) [inline tests/check-http-method.spec.js]: `done` doesn't requried in this case.

File: tests/check-http-method.spec.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+describe('checkHttpMethod function ->', () => {
+    HTTP_METHODS.forEach(method => {
+        it(`should treat HTTP method "${method}" as a correct method`, () => {
+            const result = checkHttpMethod(method);
+            expect(result).toBe(true);
+        });
+    });
+
+    it(`should be able to detect an incorrect http method`, () => {
```
Comment (JasonRammoray) [inline tests/check-http-method.spec.js]: Just one last thing.
I guess we don't need template strings in this case since we don't use any variables.

Comment (whatdoyouseedoc) [reply tests/check-http-method.spec.js]: Sure.

**Review by JasonRammoray — APPROVED** [review]

**Comment by JasonRammoray** [comment]
Published version 1.0.7.
